lib/url.js

```diff
@@ -545,135 +413,131 @@ Url.prototype.resolveObject = function(relative) {
   if (psychotic) {
     result.hostname = '';
     result.port = null;
-    if (result.host) {
-      if (srcPath[0] === '') srcPath[0] = result.host;
-      else srcPath.unshift(result.host);
+    if (!this._isEmpty(result.host)) {
+      if (srcPath[0] === '')
+        srcPath[0] = result.host;
+      else
+        srcPath.unshift(result.host);
     }
     result.host = '';
-    if (relative.protocol) {
+    if (!this._isEmpty(relative.protocol)) {
       relative.hostname = null;
       relative.port = null;
       if (relative.host) {
-        if (relPath[0] === '') relPath[0] = relative.host;
-        else relPath.unshift(relative.host);
+        if (relPath[0] === '')
+          relPath[0] = relative.host;
+        else
+          relPath.unshift(relative.host);
       }
-      relative.host = null;
+      relative.host = '';
     }
     mustEndAbs = mustEndAbs && (relPath[0] === '' || srcPath[0] === '');
   }
 
   if (isRelAbs) {
     // it's absolute.
-    result.host = (relative.host || relative.host === '') ?
-                  relative.host : result.host;
-    result.hostname = (relative.hostname || relative.hostname === '') ?
-                      relative.hostname : result.hostname;
+    result.host = !this._isEmpty(relative.host) ? relative.host : result.host;
+    result.hostname =
+        !this._isEmpty(relative.hostname) ? relative.hostname : result.hostname;
     result.search = relative.search;
     result.query = relative.query;
     srcPath = relPath;
-    // fall through to the dot-handling below.
+    // Fall through to the dot-handling below.
   } else if (relPath.length) {
-    // it's relative
+    // It's relative
     // throw away the existing file, and take the new path instead.
-    if (!srcPath) srcPath = [];
```

Comment:
A lot of seemingly unnecessary formatting changes clogging up the diff, e.g. these inserting newlines or the capitalizing of the comments. I don't mind but maybe others do.